### PR TITLE
NN-4165 pagination for view residential location

### PR DESCRIPTION
**Pull request by timoj58**

### CODE REVIEW

File: backend/util/pagination.test.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      const paginationResult = pagination(pageSize, pageOffset, total, url)
+      expect(paginationResult.results.from).toEqual(expectedFrom)
+      expect(paginationResult.results.count).toEqual(total)
+      expect(paginationResult.results.to).toEqual(expectedTo)
+      expect(paginationResult.next.href).toEqual(expectedNext)
+      expect(paginationResult.previous.href).toEqual(expectedPrevious)
+      if (expectedPages > 1) {
+        expect(paginationResult.items.length).toEqual(expectedPages)
+        paginationResult.items.forEach((item, idx) => {
+          expect(item.href).toEqual(`${url}&page=${idx + 1}`)
```
Comment (sp-watson) [inline backend/util/pagination.test.js:23]: Just a small thing, but there is no assertion for the item text or whether it is selected.

File: backend/services/allocationService.js [hunk 2]
```diff
@@ -266,6 +269,44 @@ const serviceFactory = (elite2Api, keyworkerApi, offenderSearchResultMax) => {
     }
   }
 
+  const searchOffendersPaginated = async (context, { agencyId, keywords, locationPrefix, pageRequest }) => {
+    const availableKeyworkers = await keyworkerApi.availableKeyworkers(context, agencyId)
+    log.debug('Response from available keyworker request')
```
Comment (sp-watson) [inline backend/services/allocationService.js:274]: As below - we ought to not have debug statements unless they add something useful.

I know there are some dotted around - we ought to get rid of them unless we think they are useful, but that can be done some other time.

File: backend/services/allocationService.js [hunk 3]
```diff
@@ -266,6 +269,44 @@ const serviceFactory = (elite2Api, keyworkerApi, offenderSearchResultMax) => {
...
+
+    const offenderNumbers = getOffenderNumbers(offenders)
+    const offenderKeyworkers = await keyworkerApi.offenderKeyworkerList(context, agencyId, offenderNumbers)
+
+    return {
+      keyworkerResponse: availableKeyworkers,
+      totalRecords,
+      pageOffset,
+      offenderResponse:
+        offenders.length > 0
```
Comment (sp-watson) [inline backend/services/allocationService.js:299]: No need for this as it is checked above

File: backend/services/allocationService.js [hunk 4]
```diff
@@ -266,6 +269,44 @@ const serviceFactory = (elite2Api, keyworkerApi, offenderSearchResultMax) => {
     }
   }
 
+  const searchOffendersPaginated = async (context, { agencyId, keywords, locationPrefix, pageRequest }) => {
```
Comment (sp-watson) [inline backend/services/allocationService.js:272]: It's rubbish that we don't have any existing tests for these functions (unless I am missing something), but the int tests should cover what we are adding, yes? If not, we ought to add tests.

File: backend/services/allocationService.js (hunk 2 above)
Comment (timoj58) [reply backend/services/allocationService.js:274]: sure will remove, was copied from the original method.  Will remove from those too

File: backend/util/pagination.test.js (hunk 1 above)
Comment (sp-watson) [reply backend/util/pagination.test.js:41]: Well, I would check all the others aren't selected as well, but not worth fixing right now

**Review by sp-watson — DISMISSED** [review]

**Review by sp-watson — APPROVED** [review]

**Pull request merged**